File: SearchSortLab/SearchSortLab/src/Searches.java


public class Searches
{
    public void sequentialSearch(int[] arrayToSearch, int numberToFind)
    {
        for (int i = 0; i < arrayToSearch.length; i++)
        {
               if(arrayToSearch[i] == numberToFind)
               {
                   System.out.println("Found " + numberToFind + " at index " + i);
                   return;
               }
        }
        System.out.println("Did not find " + numberToFind);
    }


    public int binarySearch(int[] arrayToSort, int key)
    {
        int low = 0;
        int high = arrayToSort.length - 1;
        int middle = (low + high + 1)/2;
        int location = -1;

        do
        {
            if(key == arrayToSort[middle])
            {
                location = middle;
            }
            else if(key < arrayToSort[middle])
            {
                high = middle - 1;
            }
            else
            {
                low = middle + 1;
            }

            middle = (low + high + 1) / 2;

        }  while((low <= high) && (location == -1));
        
        return location;
    }

    public static String leftoverNum(int[] arrayNum, int low, int high)
    {
        StringBuilder temp = new StringBuilder();

        for(int i = 0; i < low; i++)
        {
            temp.append("     ");
        }

        for(int i = 0; i <= high; i++)
        {
            temp.append(arrayNum[i] + " ");
        }

        return String.format("%s%n", temp);
    }



}

File: SortsLab/Searches.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SearchesAndSorts
{
    class Searches
    {
        public void SequentialSearch(int[] arrayToSearch, int numberToFind)
        {
            int comparisons = 0;
            for (int i = 0; i < arrayToSearch.Length; i++)
            {
                comparisons += 1;
                if (arrayToSearch[i] == numberToFind)
                {
                    Console.WriteLine("Found " + numberToFind + " at index " + i);
                    Console.WriteLine("\nComparisons in the Sequential Search: " + comparisons);
                    return;
                }
            }
            Console.WriteLine("Did not find " + numberToFind);
        }
    }
}
